core/arch/arm/kernel/generic_boot.c

```diff
@@ -589,10 +578,112 @@ static void init_fdt(unsigned long phys_fdt)
 		panic();
 	}
 }
+
+/*
+ * Check if the /secure-chosen node in the DT contains an stdout-path value
+ * for which we have a compatible driver. If so, switch the console to
+ * this device.
+ */
+static void configure_console_from_dt(unsigned long phys_fdt)
+{
+	const struct dt_driver *drv, *found = NULL;
+	const struct serial_driver *sdrv;
+	const struct fdt_property *prop;
+	struct serial_chip *dev;
+	char *stdout_data;
+	const char *uart;
+	const char *parms = NULL;
+	void *fdt;
+	int offs;
+	char *p;
+	int st;
+
+	if (!phys_fdt)
+		return;
+	fdt = phys_to_virt(phys_fdt, MEM_AREA_IO_NSEC);
+	if (!fdt)
+		panic();
+
+	offs = fdt_path_offset(fdt, "/secure-chosen");
+	if (offs < 0)
+		return;
+	prop = fdt_get_property(fdt, offs, "stdout-path", NULL);
+	if (!prop) {
+		/*
+		 * /secure-chosen node present but no stdout-path property
+		 * means we don't want any console output
+		 */
+		IMSG("Switching off console");
+		register_serial_console(NULL);
```

Comment:
Note that if `/secure-chosen` does not exist in the DT, the function does nothing (see a few lines above).

Also, there could be several UART devices with `secure-status = okay`, which one would you use as the console? This is really the job of the `/[secure-]chosen` node: pass some configuration parameters from the firmware to the [secure] OS. In addition, there might be additional configuration parameter (baud rate, parity) which really belong in `/[secure-]chosen/stdout-path`. This is well documented already (https://www.kernel.org/doc/Documentation/devicetree/bindings/chosen.txt).